### PR TITLE
Dagre: Updating Graph, adding GraphOptions

### PR DESCRIPTION
**Pull request by rustedgrail**
Cleaning up white space

Please fill in this template.

- [X] Use a meaningful title for the pull request. Include the name of the package modified.
- [X] Test the change in your own code. (Compile and run.)
- [X] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [X] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [X] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If changing an existing definition:
- [X] Provide a URL to documentation or source code which provides context for the suggested changes: <<https://github.com/cpettitt/dagre/blob/master/dist/dagre.js#L3509>>

### CODE REVIEW

File: types/dagre/index.d.ts [hunk 1]
```diff
@@ -12,15 +12,27 @@ export namespace graphlib {
         nodes(): string[];
         node(id: any): any;
         setDefaultEdgeLabel(callback: () => void): Graph;
-        setEdge(sourceId: string, targetId: string, options?: { [key: string]: any }): Graph;
+        setDefaultNodeLabel(callback: () => void): Graph;
+        setEdge(sourceId: string, targetId: string, options?: { [key: string]: any }, value?: string): Graph;
         setGraph(options: { [key: string]: any }): Graph;
         setNode(id: string, node: { [key: string]: any }): Graph;
+        graph(): GraphOptions;
```
Comment (rbuckton) [inline types/dagre/index.d.ts]: According to the linked source, `graph()` returns the value passed to `setGraph()`. It seems like the return type for `graph()` should be the same as the parameter type for `setGraph()`. Perhaps `GraphOptions` below should be `GraphLabel` and should include an indexer.

File: types/dagre/index.d.ts [hunk 2]
```diff
@@ -12,15 +12,27 @@ export namespace graphlib {
         nodes(): string[];
         node(id: any): any;
         setDefaultEdgeLabel(callback: () => void): Graph;
-        setEdge(sourceId: string, targetId: string, options?: { [key: string]: any }): Graph;
+        setDefaultNodeLabel(callback: () => void): Graph;
```
Comment (rbuckton) [inline types/dagre/index.d.ts]: From the source it looks like `setDefaultNodeLabel` can take either a [constant value _or_ a function](https://github.com/cpettitt/dagre/blob/master/dist/dagre.js#L3588), and that the function has a return value that is [non-`void`](https://github.com/cpettitt/dagre/blob/master/dist/dagre.js#L3635).

File: types/dagre/index.d.ts [hunk 3]
```diff
@@ -12,15 +12,27 @@ export namespace graphlib {
...
+        setEdge(sourceId: string, targetId: string, options?: { [key: string]: any }, value?: string): Graph;
```
Comment (rbuckton) [inline types/dagre/index.d.ts]: It seems like `setEdge` has another [overload](https://github.com/cpettitt/dagre/blob/master/dist/dagre.js#L3795).

File: types/dagre/index.d.ts [hunk 4]
```diff
@@ -12,15 +12,27 @@ export namespace graphlib {
...
         setGraph(options: { [key: string]: any }): Graph;
         setNode(id: string, node: { [key: string]: any }): Graph;
+        graph(): GraphOptions;
+
+        constructor(opt?: {});
```
Comment (rbuckton) [inline types/dagre/index.d.ts]: Do you intend to fill out the [options this constructor accepts](https://github.com/cpettitt/dagre/blob/master/dist/dagre.js#L3510)?

**Review by rbuckton — CHANGES_REQUESTED** [review]

**Comment by RyanCavanaugh** [comment]
@rustedgrail Please fix the failures indicated in the Travis CI log.

**Comment by andy-ms** [comment]
@rustedgrail Please run `tsc` in the `dagre` directory to see the new test failure.

**Comment by andy-ms** [comment]
Thanks!

**Pull request merged**